File: frontend/pms_web/src/lib/repositories/staff.ts
import apiClient from "@/lib/api/client";

export interface Staff {
  id: string;
  full_name: string;
  email: string;
  phone?: string | null;
  role: "pharmacy_admin" | "pharmacist" | "cashier";
  is_active: boolean;
  created_at: string;
}

export interface StaffCreate {
  full_name: string;
  email: string;
  phone?: string;
  role: "pharmacy_admin" | "pharmacist" | "cashier";
  password: string;
}

export const staffRepo = {
  list: async (): Promise<Staff[]> => {
    const { data } = await apiClient.get("/v1/staff");
    return data.items;
  },
  create: async (body: StaffCreate): Promise<Staff> => {
    const { data } = await apiClient.post("/v1/staff", body);
    return data;
  },
  update: async (id: string, body: Partial<StaffCreate & { is_active: boolean }>) => {
    const { data } = await apiClient.patch(`/v1/staff/${id}`, body);
    return data as Staff;
  },
};

File: backend/services/pms_service/app/schemas/staff.py
from __future__ import annotations

from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict

Role = Literal["pharmacy_admin", "pharmacist", "cashier"]


class StaffCreate(BaseModel):
    full_name: str
    email: str
    phone: str | None = None
    role: Role
    password: str


class StaffUpdate(BaseModel):
    full_name: str | None = None
    phone: str | None = None
    role: Role | None = None
    is_active: bool | None = None
    password: str | None = None


class StaffOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    full_name: str
    email: str
    phone: str | None
    role: Role
    is_active: bool
    created_at: datetime


class StaffList(BaseModel):
    items: list[StaffOut]
